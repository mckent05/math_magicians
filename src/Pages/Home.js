import React from 'react';

const Home = () => (
  <div className="home-page">
    <h2>Welcome To Maths Magicians</h2>
    <p>
      <span>This is a single Page Application for all maths-lovers </span>
      <span>this page allows mathematicians perform some of their daily and basic calculation</span>
      <span> exrercises and also read a Mathematics quote. We hope you enjoy it!!!</span>
    </p>
  </div>
);

export default Home;
