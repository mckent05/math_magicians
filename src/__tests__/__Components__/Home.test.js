import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import Home from '../../Pages/Home';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('testing for the UI of HomePage', () => {
  act(() => {
    render(<Home />, container);
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
