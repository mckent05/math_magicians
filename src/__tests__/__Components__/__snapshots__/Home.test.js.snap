// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for the UI of HomePage 1`] = `
"<div class=\\"home-page\\">
  <h2>Welcome To Maths Magicians</h2>
  <p><span>This is a single Page Application for all maths-lovers </span><span>this page allows mathematicians perform some of their daily and basic calculation</span><span> exrercises and also read a Mathematics quote. We hope you enjoy it!!!</span></p>
</div>"
`;
